src/marqo/tensor_search/tensor_search.py

```diff
@@ -1235,6 +1346,257 @@ def _lexical_search(
+    vector_job = VectorisedJobs(
+        model_name=index_info.model_name,
+        model_properties=_get_model_properties(index_info),
+        content=content,
+        device=device,
+        normalize_embeddings=index_info.index_settings['index_defaults']['normalize_embeddings'],
+        image_download_headers=q.image_download_headers
+    )
+    # If exists, add content to vector job. Otherwise create new
+    if jobs.get(vector_job.groupby_key()) is not None:
+        j = jobs.get(vector_job.groupby_key())
+        ptr = j.add_content(content)
+    else:
+        jobs[vector_job.groupby_key()] = vector_job
+        ptr = VectorisedJobPointer(
+            job_hash=vector_job.groupby_key(),
+            start_idx=0,
+            end_idx=len(vector_job.content)
+        )
+    return ptr
+
+def _bulk_vector_text_search(config: Config, queries: List[BulkSearchQueryEntity], device=None, raise_on_searchable_attribs=False, simplified_format=True, number_of_highlights: int = 3, return_doc_ids=False,  result_count: int = 5):
+    if len(queries) == 0:
+        return []
+
+    start_preprocessing_time = timer()
+    jobs: Dict[JHash, VectorisedJobs] = {} # job_hash to job
+    qidx_to_job: Dict[Qidx, VectorisedJobPointer] = {}
+    selected_device = config.indexing_device if device is None else device
+
+    ## 1. Pre-process inputs ready for s2_inference.vectorise
+    for i in range(len(queries)):
+        q = queries[i]
+        index_info = get_index_info(config=config, index_name=q.index)
+        to_be_vectorised: List[List[str]] = construct_vector_input_batches(q.q, index_info)
+        qidx_to_job[i] = assign_query_to_vector_job(q, jobs, to_be_vectorised, index_info, selected_device)
+
+    ## 2. Vectorise against all queries
+    job_ptr_to_vectors: Dict[JHash, List[float]] = {}
+    for idx, v in jobs.items():
+        job_ptr_to_vectors[idx] = functools.reduce(
+            lambda x, y: x + y,
+            [s2_inference.vectorise(
+                model_name=v.model_name, model_properties=v.model_properties,
+                content=batch, device=v.device,
+                normalize_embeddings=v.normalize_embeddings,
+                image_download_headers=v.image_download_headers
+            ) for batch in v.content] # Todo: check if should be batching.
+        )
+
+    ## 3. Get vectors back from batch, process `ordered_queries` into each query.
+    qidx_to_vectors: Dict[Qidx, List[List[float]]] = defaultdict(list)
+    for qidx, ptr in qidx_to_job.items():
+        vectors = job_ptr_to_vectors[ptr.job_hash][ptr.start_idx: ptr.end_idx]
+        qidx_to_vectors[qidx].append(vectors)
+
+        q = queries[qidx]
+        ordered_queries = list(q.q.items()) if isinstance(q.q, dict) else None
+        if ordered_queries:
+            # multiple queries. We have to weight and combine them:
+            index_info = get_index_info(config=config, index_name=q.index)
```

Comment:
this isn't a blocking call to _mappings, right?

Reply:
So we run `fill_index_cache` at the start of `bulk_search` for all the indices in the search requests. I think this should be sufficient to ensure `get_index_info` should be in cache.